crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1394,14 +1453,19 @@ impl<'db> ClassType<'db> {
     }
 
     pub fn inherited_class_member(self, db: &'db dyn Db, name: &str) -> Type<'db> {
-        for base in self.bases(db) {
-            let member = base.member(db, name);
-            if !member.is_unbound() {
-                return member;
+        let mro_possibilities = self.mro_possibilities(db);
+        let mut union_builder = UnionBuilder::new(db);
+        'outer: for mro in mro_possibilities.iter(db, mro::ClassBase::Class(self)) {
+            for base in &*mro {
+                let member = base.own_class_member(db, name);
+                if !member.is_unbound() {
+                    union_builder = union_builder.add(member);
+                    continue 'outer;
```

Comment:
Nit: Isn't this the same as using `break`?
```suggestion
                    break;
```

Reply:
No, we need to skip the `union_builder = union_builder.add(Type::Unbound);` line which is called after the termination of the inner loop. We wouldn't skip that line if we used `break`.